sycl/ReleaseNotes.md

```diff
@@ -1,3 +1,393 @@
+- Fixed race conditions in the group algorithms implemented in libclc for AMDGPU because the control barriers were not emitting any fences due to unspecified semantics. intel/llvm#12873
+- Fixed device assertion bug on Windows where the assertion message is not printed when an assertion is triggered. intel/llvm#15232
+- Made `event::get_backend()`, `size()` method of the image classes, `device_image::has_kernel()` to be exception-free as they are `noexcept` according to specification. intel/llvm#15173
+- Fixed linker errors for `WorkGroup` collective functions on Native CPU backend. intel/llvm#15144
+- Fixed a hang when invalid values are provided to `ONEAPI_DEVICE_SELECTOR`. intel/llvm#15255
+- Addressed several issues in GDB xmethods: printing non-sycl types may generate a python exception, accessor subscripts using `size_t` report an unsupported subscript type, multi-dimensional accessors calculate the wrong array layout. intel/llvm#15250
+- Fixed a flaky failure when getting write access to a buffer from multiple threads. intel/llvm#15273
+- Fixed `nullptr` input handling for `make_kernel_bundle` interoperability API. intel/llvm#15247
+- Enabled usage of Windows proxy loader for UR. After switch from PI to UR, SYCL-RT linked with `ur_loader.dll`directly and still  experienced issues with race conditions in the teardown of SYCL-RT and  Unified Runtime. This change is intented to resolve this issue. intel/llvm#15262
+- Fixed "out of device memory" error handling in the Program Manager. intel/llvm#15335
+- Fixed handling of interop events for barrier with waitlist. intel/llvm#15352
+- Fixed image selection for AOT on `intel_cpu_{spr, gnr}`. intel/llvm#15208
+- Added missing synchronization for host task after queue barrier. intel/llvm#15345
+- Fixed handling of extensions that OpenCL FPGA driver doesn't report as supported, but which are actually supported (at least to the extent that UR and SYCL require). intel/llvm#15350
+- Fixed possible race at XPTI initialization in UR to resolve the issue with the `ur.call` XPTI stream being not be visible.  intel/llvm#15367
+- Fixed the issue for the case when `ext_oneapi_barrier` fails and runtime incorrectly trying to release invalid event handle. intel/llvm#15367
+- Resolved an overload resolution ambiguity on Windows self-builds when using the unary minus operator with bfloat16 by taking a const-qualified argument. intel/llvm#15393
+- Fixed the behavior of `static|dynamic_address_cast` when target address space is `generic`. intel/llvm#15394
+- Fixed behaviour of the queue barrier with waitlist for the in-order lists mode in the L0 adapter. intel/llvm#15404
+- Fixed in-order queue dependencies for no-scheduler path when multiple queues are used. intel/llvm#15412
+- Fixed queue barrier with waitlist when used with interoperability events. intel/llvm#15488
+- Fixed edge cases for `exp(complex)` in device code. intel/llvm#15489  intel/llvm#15672 intel/llvm#15980 intel/llvm#15808
```

Comment:
There is a line about `exp(complex)` fix on Windows, let's merge it here as a one more PR with those fixes